src/ethereum/frontier/vm/instructions/sha3.py

```diff
@@ -0,0 +1,59 @@
+"""
+Ethereum Virtual Machine (EVM) SHA3 Instructions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. contents:: Table of Contents
+    :backlinks: none
+    :local:
+
+Introduction
+------------
+
+Implementations of the EVM sha3 instructions.
+"""
+
+from ethereum.base_types import U256, Uint
+from ethereum.crypto import keccak256
+from ethereum.utils import get_ceil32
+
+from .. import Evm
+from ..gas import GAS_KECCAK256, GAS_KECCAK256_WORD, subtract_gas
+from ..memory import expand_memory, get_words
+from ..stack import pop, push
+
+
+def sha3(evm: Evm) -> None:
+    """
+    Pushes to the stack the Keccak-256 hash of a region of memory.
+
+    This also expands the memory in multiples of words (32 bytes), in case
+    the memory is insufficient to access the word's memory location.
+
+    Parameters
+    ----------
+    evm :
+        The current EVM frame.
+
+    Raises
+    ------
+    StackUnderflowError
+        If `len(stack)` is less than `2`.
+    """
+    # Converting memory_start_index to Uint and memory_end_index can
+    # overflow U256.
+    memory_start_index = Uint(pop(evm.stack))
```

Comment:
It feels like converting to `Uint` would be clearer if done exactly where needed? E.g. below could be `required_mem_size = Uint(mem_start_idx) + Uint(data_len)` - this way it's in the same line it happens rather than keeping in mind you're dealing with `Uints`?

Reply:
Hi, in the latest changes after rebasing on top of the memory operations, only `memory_start_index` is type-casted as `Uint` and it is to be used as `Uint` throughout the code. Please take a look and let me know if you still prefer to change it. Thanks.